semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -374,6 +374,20 @@ let rec m_name a b =
       m_name a (B.Id (idb, B.empty_id_info ()))
       >||> (* try this time a match with the resolved entity *)
       m_name a (H.name_of_ids dotted)
+      >||>
+      (* Try the parents *)
```

Comment:
next time please add a (* deep: *) comment so we know all the places relevant to deep semgrep.
(It's usually the same that the places where we use !hook_xxx, but still, I like tagging those places).